tests/e2e/driver_tests.py

```diff
@@ -360,6 +360,42 @@ def execute_really_long_query():
             cursor.execute("SELECT * FROM range(3)")
             self.assertEqual(len(cursor.fetchall()), 3)
 
+    def test_retry_after_connection_timeout(self):
+        
+        import socket
+        from databricks.sql.thrift_api.TCLIService import ttypes
+
+        ATTEMPTS_TO_TRY = 2
+        
+        with self.cursor({}) as cursor:
+            cursor.execute("SELECT id FROM RANGE(10)")
+            op_handle = cursor.active_op_handle
+            thrift_backend = cursor.thrift_backend
+        
+
+        thrift_backend._retry_stop_after_attempts_count = ATTEMPTS_TO_TRY
+        req = ttypes.TGetOperationStatusReq(
+            operationHandle=op_handle,
+            getProgressUpdate=False,
+        )
+
+        
+        with self.assertRaises(OperationalError) as cm:
```

Comment:
I actually removed the e2e tests for this behaviour because they were no more useful than unit tests. I'll add an e2e test for this scenario after we merge support for http proxies. Then we can simulate real timeouts.

Reply:
In the unit tests asserts on `RequestError` btw.